google/cloud/storage/examples/storage_quickstart.cc

```diff
@@ -31,7 +31,14 @@ int main(int argc, char* argv[]) try {
 
   // Create a client to communicate with Google Cloud Storage. This client
   // uses the default configuration for authentication and project id.
-  gcs::Client client;
+  google::cloud::StatusOr<gcs::Client> status_or_client =
+      gcs::Client::CreateDefaultClient();
+  if (!status_or_client.ok()) {
+    std::cerr << "Failed to create Storage Client, status="
+              << status_or_client.status() << std::endl;
+    return 1;
+  }
+  gcs::Client client = std::move(status_or_client.value());
```

Comment:
Generally we prefer to use the `StatusOr<T>` directly, as a smart pointer, and I believe (but @devjgm can correct me here), that we prefer:

```C++
  gcs::Client client = *std::move(status_or_client);
```

if we are going to transfer the contents of the `StatusOr<>` somewhere else.

Reply:
I agree with what @coryan said here. And to add a tiny bit more...

In general it's easier and more efficient to avoid the move completely and directly access the object contained inside the StatusOr. StatusOr provides smart-pointer like syntax in the form of operator-> to do this. This works well in the vast majority of situations. If there is a situation where you do need to transfer the contained object somewhere else, a std::move like @coryan mentioned is probably best.

Reply:
Used it in pointer-style; fixed.